src/Integration/ProjectPropertyManager.cs

```diff
@@ -0,0 +1,123 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ProjectPropertyManager.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using EnvDTE;
+using System;
+
+namespace SonarLint.VisualStudio.Integration
+{
+    [Export(typeof(IProjectPropertyManager))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class ProjectPropertyManager : IProjectPropertyManager
```

Comment:
This is a MEF component which consolidates the parsing of property strings into `bool`s and `bool?`s, for specific project properties. It largely wraps the `IProjectSystemHelper` calls because it is a 'local service' and to get access to it from the Integration.Vsix project, I'd have to make `IHost` public.. which is turns means I have to make a whole host of other classes public, including `ProjectViewModel` and `ServerViewModel`, which seems to indicate an larger arch. issue.